src/MoBi.Presentation/Presenter/SelectBuildingBlocksForExtendPresenter.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.Linq;
+using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Domain.Repository;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Presentation.Presenters;
+
+namespace MoBi.Presentation.Presenter
+{
+   public interface ISelectBuildingBlocksForExtendPresenter : IDisposablePresenter
+   {
+      IEnumerable<MoleculeBuildingBlock> GetMolecules();
+      IEnumerable<MoBiSpatialStructure> GetSpatialStructures();
+
+      void SelectBuildingBlocksForExtend(bool moleculeRequired = true);
+      MoBiSpatialStructure SelectedSpatialStructure { get; }
+      MoleculeBuildingBlock SelectedMoleculeBuildingBlock { get; }
+   }
+
+   public class SelectBuildingBlocksForExtendPresenter : AbstractDisposablePresenter<ISelectSpatialStructureAndMoleculesView, ISelectBuildingBlocksForExtendPresenter>, ISelectBuildingBlocksForExtendPresenter
+   {
+      private readonly IBuildingBlockRepository _buildingBlockRepository;
+      private SelectSpatialStructureAndMoleculesDTO _dto;
+
+      public SelectBuildingBlocksForExtendPresenter(ISelectSpatialStructureAndMoleculesView view, IBuildingBlockRepository buildingBlockRepository) : base(view)
+      {
+         _buildingBlockRepository = buildingBlockRepository;
+      }
+
+      public void SelectBuildingBlocksForExtend(bool moleculeRequired = true)
+      {
+         setViewCaption(moleculeRequired);
+         _dto = createDTO(GetMolecules().FirstOrDefault(), GetSpatialStructures().FirstOrDefault(), moleculeRequired);
+         _view.Show(_dto);
+         _view.Display();
+         if (!_view.Canceled)
+            return;
+
+         _dto.Molecules = null;
+         _dto.SpatialStructure = null;
+      }
+
+      private void setViewCaption(bool moleculeRequired)
+      {
+         _view.Caption = moleculeRequired ? AppConstants.Captions.SelectSpatialStructureAndMolecules : AppConstants.Captions.SelectSpatialStructure;
+      }
+
+      private SelectSpatialStructureAndMoleculesDTO createDTO(MoleculeBuildingBlock moleculeBuildingBlock, MoBiSpatialStructure spatialStructure, bool moleculeRequired)
+      {
+         var dto = new SelectSpatialStructureAndMoleculesDTO(moleculeRequired)
+         {
+            Molecules = moleculeBuildingBlock,
+            SpatialStructure = spatialStructure
+         };
+         return dto;
+      }
+
+      public MoBiSpatialStructure SelectedSpatialStructure => _dto.SpatialStructure;
+      public MoleculeBuildingBlock SelectedMoleculeBuildingBlock => _dto.Molecules;
```

Comment:
from a cohesion stand point, we have two different ways of implementing the same thing. one is => the other one is 
{

xx

}}

Let's make ti consistent by inlining  public IEnumerable<MoleculeBuildingBlock> GetMolecules()

and in fact, this should be a get as well. Why is this not a get like SelectedSpatialStructure?

Reply:
and also the other one